### DISCUSSION TITLE
Streaming in Next.js (Pages) API routes not working

### DISCUSSION THREAD

**Discussion by giuseppeg** [vercel/next.js#67026] [Help] [answered]
### Summary

**Update**: Vercel confirmed that the docs were wrong and streaming is not supported in the Next.js pages.

I have a Next.js (**pages**) API route (**Node.js handler**) and want to stream some data but streaming doesn't work.

According to https://vercel.com/docs/functions/streaming#streaming-functions it should just work.

I am setting the route config `supportsResponseStreaming` flag to `true` but streaming doesn't work on Vercel. It however seems to be working locally (`npm run dev` or `vercel dev`).

### Additional information

On Vercel the API route buffers the response and flushes only at the end when the connection is closed with `res.end()`.

### Example

Source: https://gist.github.com/giuseppeg/ae8150a53c5caea8b88f8283ea63b6d3
Deployment: https://test-node-streaming-seven.vercel.app

**Comment by SalmenBejaoui** [comment, marked as answer]
The streaming works only with the App Router.

**Reply by giuseppeg** [reply]
Yes that seems the case but this is a pages router app and issue and I would like to figure out why streaming doesn't work there. 

The docs suggest that the Node js handler in pages router should work when the `supportsResponseStreaming` is `true` https://vercel.com/docs/functions/streaming#streaming-functions

and there is a discrepancy between development and production (Vercel). It'd be great if someone from the team could clarify as I am trying to avoid a refactoring if possible.

Perhaps it indeed works only with App Router but the docs are wrong or out to date?

![vercel-streaming-1](https://github.com/vercel/next.js/assets/711311/74a9508d-b64b-426d-b9f4-4c8ecad2c564)

**Reply by giuseppeg** [reply]
Thanks @SLMNBJ. Vercel confirmed that the docs were wrong and streaming is not supported in the Next.js pages.
